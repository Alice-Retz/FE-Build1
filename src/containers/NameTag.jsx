import React, { Component } from 'react';
import Controls from '../components/name-tag/Controls';
import Display from '../components/name-tag/Display';

export default class NameTag extends Component {
    state = {
      greeting: '',
      tagName: '',
      newName: '',
      newGreeting: '',
    };

    handleNameChange = (e) => {
      this.setState({ tagName: e.target.value });
    };
    handleGreetChange = (e) => {
      this.setState({ greeting: e.target.value });
    };
    handleSubmit = (e) => {
      e.preventDefault();
      this.setState({
        newName: this.state.tagName,
        newGreeting: this.state.greeting,
      });
    };

    render() {
      const { tagName, greeting, newName, newGreeting } = this.state;
      return (
        <>
          <Controls
            greeting={greeting}
            onNameChange={this.handleNameChange}
            tagName={tagName}
            onGreetChange={this.handleGreetChange}
            onSubmit={this.handleSubmit}
          />
          <Display newName={newName} newGreeting={newGreeting} />
        </>
      );
    }
}
