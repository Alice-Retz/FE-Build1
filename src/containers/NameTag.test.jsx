import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import NameTag from '../containers/NameTag';

describe('NameTag container', () => {
  it('changes the displayed name', async () => {
    render(<NameTag />);
    screen.getByText('my name is');
    const nameInput = await screen.findByLabelText('Name:');
    userEvent.type(nameInput, 'Alice');
    const greetingInput = await screen.findByLabelText('Greeting:');
    userEvent.type(greetingInput, 'Hello');
    const button = await screen.findByRole('button', { name: 'button' });
    userEvent.click(button);
    return waitFor(() => {
      const greeting = screen.getAllByText('Hello', { exact: false, });
      const name = screen.getAllByText('Alice', { exact: false, });
      expect(greeting).toMatchSnapshot();
      expect(name).toMatchSnapshot();
    });
  });
});
