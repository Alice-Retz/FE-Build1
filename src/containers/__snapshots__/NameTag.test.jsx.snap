// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameTag container changes the displayed name 1`] = `
Array [
  <h1>
    Hello
    ,
  </h1>,
]
`;

exports[`NameTag container changes the displayed name 2`] = `
Array [
  <h1>
    Alice
  </h1>,
]
`;
